iModelJsNodeAddon/api_package/ts/src/test/DgnDb.test.ts

```diff
@@ -31,6 +31,44 @@ describe("basic tests", () => {
     done();
   });
 
+  it("compress/decompress", () => {
+    const assertCompressAndThenDecompress = (sourceData: Uint8Array) => {
+      const compressData = iModelJsNative.DgnDb.zlibCompress(sourceData);
+      const decompressData = iModelJsNative.DgnDb.zlibDecompress(compressData, sourceData.length);
+      assert.deepEqual(sourceData, decompressData);
+    };
+    const randomData = (n: number) => {
+      return Uint8Array.from(Array.from({ length: n }, () => Math.floor(Math.random() * 10)));

```

Comment:
Please don't use random data in tests.

Reply:
Yes, I thought about it and agree not to use random numbers. I had difficulty compressing a small array as it resulted in a bigger buffer than the source buffer and did not want to paste a large int array in test code. So I used random but generated a less dense array as random numbers are just between 0-10 and not 0-255. This allows data to be always compressed significantly as its zlib which uses Huffman compression (assigning small code to more frequent characters)

But now that I think it could be just an array of single numbers [1,1,1...] which will be even better as the buffer can be small and the encoder will just assign a bit to each digit.

Let me fix it.

Reply:
Out of curiosity, what is the concern with random data? Some permutation of data might cause problems?

Reply:
Tests should be deterministic. Random data introduces potential flakiness and even if flakiness is impossible, it's better to be explicit about your inputs and expected outputs. It's fine to generate a random data set ahead of time and then hard-code that as your test data - just don't generate new random data each time the test runs.

Reply:
Just to avoid pasting the 1K array into the test file. Any compression algorithm poorly compresses a randomized buffer as it takes advantage of patterns in data. With no repeating pattern, it cannot compress. So when I use a randomized buffer it has to be 1K to be compressed a little bit. But as I said I can just have to repeat data for it to be compressible.